### ISSUE TITLE
t.p.util.OrderedDict should be deprecated and replaced by the stdlib's one

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8051]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8051|
|Type|enhancement|
|Created|2015-10-19 01:58:52Z|
|Branch|https://github.com/twisted/twisted/tree/no-ordereddict-8051|

t.p.util.OrderedDict is bad and we should replace it with the one in ```collections```.

<details><summary>Searchable metadata</summary>

```
trac-id__8051 8051
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__EndOfLifePython2_6 EndOfLifePython2.6
branch__branches_no_ordereddict_8051 branches/no-ordereddict-8051
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1445219932809042 1445219932809042
changetime__1445304933694640 1445304933694640
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45832]) Branching to no-ordereddict-8051.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Tests are looking fairly good. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_3"></a>|@jonathanj set owner to @hawkowl|
|-|-|

I think this mostly looks good but removing `OrderedDict` from `twisted.python.compat` entirely seems like a breaking change, unless there is a special rule for this module shouldn't this attribute be subjected to the same deprecation policy as `twisted.python.util.OrderedDict`?

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj commented|
|-|-|

After fixing my previous comment this looks fine to merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45856]) Merge no-ordereddict-8051: Deprecate twisted.python.util.OrderedDict

Author: hawkowl
Reviewer: jonathanj
Fixes: [#8051](https://github.com/twisted/twisted/issues/8051)
